### PR TITLE
SCRUM-4090 Paralogy Load clean up

### PR DESCRIPTION
**Pull request by VarunReddy1111**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java [hunk 1]
```diff
@@ -62,4 +67,30 @@ public void execLoad(BulkLoadFile bulkLoadFile) {
 		}
 	}
 
+	private void runCleanup(BulkLoadFileHistory history, String dataProvider, List<Long> paralogyPairsBefore, List<Long> paralogyPairsAfter) {
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java]: @VarunReddy1111 can the generic cleanup be used or do we need this specific one?

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java [hunk 2]
```diff
@@ -3,25 +3,32 @@
 import java.io.FileInputStream;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - java.util.stream.Collectors.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java [hunk 3]
```diff
@@ -3,25 +3,32 @@
...
 import java.util.zip.GZIPInputStream;
 
+import org.alliancegenome.curation_api.dao.GeneToGeneParalogyDAO;
 import org.alliancegenome.curation_api.enums.BackendBulkDataProvider;
-
+import org.alliancegenome.curation_api.exceptions.ObjectUpdateException.ObjectUpdateExceptionData;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - org.alliancegenome.curation_api.exceptions.ObjectUpdateException.ObjectUpdateExceptionData.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java [hunk 4]
```diff
@@ -3,25 +3,32 @@
...
 import org.alliancegenome.curation_api.interfaces.AGRCurationSchemaVersion;
 import org.alliancegenome.curation_api.model.entities.GeneToGeneParalogy;
 import org.alliancegenome.curation_api.model.entities.bulkloads.BulkFMSLoad;
 import org.alliancegenome.curation_api.model.entities.bulkloads.BulkLoadFile;
 import org.alliancegenome.curation_api.model.entities.bulkloads.BulkLoadFileHistory;
 import org.alliancegenome.curation_api.model.ingest.dto.fms.ParalogyIngestFmsDTO;
 import org.alliancegenome.curation_api.services.GeneToGeneParalogyService;
+import org.alliancegenome.curation_api.util.ProcessDisplayHelper;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - org.alliancegenome.curation_api.util.ProcessDisplayHelper.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java [hunk 5]
```diff
@@ -3,25 +3,32 @@
...
+import org.apache.commons.collections.ListUtils;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - org.apache.commons.collections.ListUtils.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java [hunk 6]
```diff
@@ -38,20 +45,18 @@
...
 			BulkLoadFileHistory history = new BulkLoadFileHistory(paralogyData.getData().size());
-
 			createHistory(history, bulkLoadFile);
-			runLoad(geneToGeneParalogyService, history, dataProvider, paralogyData.getData(), paralogyIdsLoaded);
-
-			// runCleanup(history, fms.getFmsDataSubType());
+			boolean success = runLoad(geneToGeneParalogyService, history, dataProvider, paralogyData.getData(), paralogyIdsLoaded);
 
+			if (success) {
+				runCleanup(geneToGeneParalogyService, history, fms.getFmsDataSubType(), paralogyPairsBefore, paralogyIdsLoaded,fms.getFmsDataType(), bulkLoadFile.getMd5Sum(),false);
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>',' is not followed by whitespace.

Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/ParalogyExecutor.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>',' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/LoadFileExecutor.java [hunk 7]
```diff
@@ -240,8 +240,12 @@ protected <E extends AuditedObject, T extends BaseDTO> boolean runLoad(BaseUpser
 		return true;
 	}
 
+	protected <S extends BaseEntityCrudService<?, ?>> void runCleanup(S service, BulkLoadFileHistory history, String dataProviderName, List<Long> annotationIdsBefore, List<Long> annotationIdsAfter, String loadTypeString, String md5sum){
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/LoadFileExecutor.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' is not preceded with whitespace.

**Review by oblodgett — APPROVED** [review]

**Review by markquintontulloch — APPROVED** [review]

**Pull request merged**